services/core/shared/downloadFile.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const fs = require('fs');
+const https = require('https');
+const tar = require('tar');
+
+const {
+	Logger,
+	Exceptions: { NotFoundException },
+	HTTP: { request },
+} = require('lisk-service-framework');
+
+const logger = Logger();
+
+const downloadAndExtractTarball = (url, directoryPath) => new Promise((resolve, reject) => {
```

Comment:
Can we also take this opportunity to implement a `downloadAndUnzipFile` method, for situations where we only have the snapshot zipped but not tarball (Eg: `*.sql.gz`)?

Ref: https://stackoverflow.com/questions/10308110/simplest-way-to-download-and-unzip-files-in-node-js-cross-platform. Please check the `zlib` module based solution.

Reply:
I have implemented the required method `downloadAndUnzipFile` and updated the code.